sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ResourceExtensions.cs

```diff
@@ -19,13 +19,15 @@ internal static class ResourceExtensions
     private const string DefaultServiceName = "unknown_service";
     private const int Version = 2;
 
-    internal static AzureMonitorResource? CreateAzureMonitorResource(this Resource resource, string? instrumentationKey = null)
+    internal static AzureMonitorResource? CreateAzureMonitorResource(this Resource resource, string? instrumentationKey = null, IPlatform ? platform = null)
     {
         if (resource == null)
         {
             return null;
         }
 
+        platform ??= DefaultPlatform.Instance;
```

Comment:
If it's only for testing, we pass the platform as a parameter. We might need to reconsider this change. Can't we just use DefaultPlatform.Instance and store it in an internal variable for testing purposes?

Reply:
No, manipulating static singletons in unit tests is going to create new problems. This is exactly what we're trying to prevent.

Reply:
discussed offline, will use method overload to clearly document that this specific scenario is for unit testing.